mitodl/bootcamp-ecommerce
# Application model and admin fixes

#### What are the relevant tickets?
No ticket

#### What's this PR do?
- Fixes/cleans up model admin for a few models
- Adds a uniqueness constraint to prevent multiple submissions for the same application step
- Adds validation to make sure that the bootcamps/runs configured for application steps and submissions are the same
- Adds factories for the new application models

#### How should this be manually tested?
- Use the admin for `BootcampRunApplicationStep` admin and set the `bootcamp_run` and `application_step` so the related bootcamps are not the same, then try to save. You should get a validation error
- Use the admin for `ApplicationStepSubmission` and set the `bootcamp_application` and `run_application_step` so the related bootcamp runs are not the same, then try to save. You should get a validation error

#### Any background context you want to provide?
This is rebased on #454
